opennms-config/src/test/java/org/opennms/netmgt/config/PollerConfigReloadIT.java

```diff
@@ -131,4 +144,93 @@ private InputStream setIncludeUrlFileInConfig(URL configUrl) throws IOException
         String modifiedString = marshalledString.replace("${INCLUDE_URL_FILE}", includeUrlFile.getAbsolutePath());
         return IOUtils.toInputStream(modifiedString, Charset.defaultCharset());
     }
+
+    static void setFinalStatic(Field field, Object newValue) throws Exception {
+        field.setAccessible(true);
+
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+
+        field.set(null, newValue);
+    }
+
+    @Test
+    public void testPollerConfigReloadFail() throws Exception {
+        final File temporaryFile = File.createTempFile("poller-configuration-", ".xml", new File(PollerConfigReloadIT.class.getResource("/etc").getFile()));
+        System.setProperty("opennms.home", temporaryFile.toPath().getParent().getParent().toString());
+
+        final AtomicBoolean invalid = new AtomicBoolean(false);
+        FilterDaoFactory.setInstance(new FilterDao() {
+            @Override
+            public SortedMap<Integer, String> getNodeMap(String rule) throws FilterParseException {
+                return null;
+            }
+
+            @Override
+            public Map<InetAddress, Set<String>> getIPAddressServiceMap(String rule) throws FilterParseException {
+                return null;
+            }
+
+            @Override
+            public void flushActiveIpAddressListCache() {
+            }
+
+            @Override
+            public List<InetAddress> getIPAddressList(String rule) throws FilterParseException {
+                return null;
+            }
+
+            @Override
+            public boolean isValid(String addr, String rule) throws FilterParseException {
+                return false;
+            }
+
+            @Override
+            public boolean isRuleMatching(String rule) throws FilterParseException {
+                return false;
+            }
+
+            @Override
+            public void validateRule(String rule) throws FilterParseException {
+                if (invalid.get()) {
+                    throw new FilterParseException("Something fishy");
+                }
+            }
+
+            @Override
+            public List<InetAddress> getActiveIPAddressList(String rule) throws FilterParseException {
+                return Lists.newArrayList();
+            }
+        });
+
+        final String fileIdToName[] = new String[83];
+        fileIdToName[8] = temporaryFile.getName();
+        setFinalStatic(ConfigFileConstants.class.getDeclaredField("FILE_ID_TO_NAME"), fileIdToName);
```

Comment:
I'd look at adding a variable and a setter to the `PollerConfigFactory` to configure this rather than messing with the `ConfigFileConstants`.

Reply:
done